pkg/adapter/incoming.go

```diff
@@ -95,20 +221,73 @@ func (l *listener) processIncoming(targetRepo *v1alpha1.Repository) (provider.In
 	l.event.Organization = org
 	l.event.Repository = repo
 
+	var isProviderTypeNotSet bool
+	if targetRepo.Spec.GitProvider == nil {
+		isProviderTypeNotSet = true
+	} else if targetRepo.Spec.GitProvider.Type == "" {
+		isProviderTypeNotSet = true
+	}
+
 	var provider provider.Interface
-	switch targetRepo.Spec.GitProvider.Type {
-	case "github":
+	if isProviderTypeNotSet {
 		provider = github.New()
-	case "gitlab":
-		provider = &gitlab.Provider{}
-	case "gitea":
-		provider = &gitea.Provider{}
-	case "bitbucket-cloud":
-		provider = &bitbucketcloud.Provider{}
-	case "bitbucket-server":
-		provider = &bitbucketserver.Provider{}
-	default:
-		return l.processRes(false, nil, l.logger, "", fmt.Errorf("no supported Git provider has been detected"))
+	} else {
+		switch targetRepo.Spec.GitProvider.Type {
+		case "github":
+			provider = github.New()
+		case "gitlab":
+			provider = &gitlab.Provider{}
+		case "gitea":
+			provider = &gitea.Provider{}
+		case "bitbucket-cloud":
+			provider = &bitbucketcloud.Provider{}
+		case "bitbucket-server":
+			provider = &bitbucketserver.Provider{}
+		default:
+			return l.processRes(false, nil, l.logger, "", fmt.Errorf("no supported Git provider has been detected"))
+		}
 	}
+
 	return l.processRes(true, provider, l.logger.With("provider", "incoming"), "", nil)
 }
+
+type JWTClaim struct {
+	Issuer int64 `json:"iss"`
+	jwt.RegisteredClaims
+}
+
+func GenerateJWT(ctx context.Context, run *params.Run) (string, error) {
+	ns := os.Getenv("SYSTEM_NAMESPACE")
```

Comment:
we already have GetappToken doing this, can we not have duplication? pkg/provider/github/parse_payload.go